backend/app/schemas/position.py

```
@@ -1,27 +1,360 @@
+
+class PositionUpdate(BaseModel):
+    """Schema for updating position fields."""
+    quantity: Optional[Decimal] = Field(None, gt=0, description="Current quantity held")
+    average_cost: Optional[Decimal] = Field(None, gt=0, description="Average cost per unit including fees")
+    cost_basis: Optional[Decimal] = Field(None, gt=0, description="Total cost basis")
+    exchange: Optional[str] = Field(None, description="Crypto exchange (for crypto assets)")
+    wallet_address: Optional[str] = Field(None, description="Wallet address (for crypto assets)")
+
+    @field_validator('quantity')
+    @classmethod
+    def validate_quantity_precision(cls, v, info):
```

Ruff (0.13.3)

Unused class method argument: `info`

(ARG003)